src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useCallback, useState, useMemo } from 'react';
+import {
+    _cs,
+} from '@togglecorp/fujs';
+
+import {
+    Container,
+    List,
+} from '@the-deep/deep-ui';
+import _ts from '#ts';
+
+import FileItem from './FileItem';
+import { FileUploadResponse } from '../types';
+import FilesSearchInput from './FilesSearchInput';
+import styles from './styles.scss';
+
+const keySelector = (d: FileUploadResponse): number => d.id;
+
+interface Props {
+    className?: string;
+    onDeleteFile: (id: number) => void;
+    files: FileUploadResponse[];
+}
+
+function FilesUploaded(props: Props) {
+    const {
+        className,
+        onDeleteFile,
+        files = [],
+    } = props;
+
+    const [searchResults, setSearchResults] = useState<number[] | undefined>();
+    const fileRendererParams = useCallback((
+        _: number,
+        data: FileUploadResponse,
+    ) => ({
+        data,
+        onDeleteFile,
+    }), [onDeleteFile]);
+
+    const handleSearchFiles = useCallback((matchingFiles: number[] | undefined) => {
+        setSearchResults(matchingFiles);
+    }, []);
+
+    const searchedFiles = useMemo(() => {
+        if (searchResults) {
+            return files.filter(f => searchResults.includes(f.id));
+        }
+        return files;
+    }, [files, searchResults]);
+
+    return (
+        <div
+            className={_cs(className, styles.filesUploadedDetails)}
+        >
+            <Container
+                className={styles.filesContainer}
+                heading={_ts('bulkUpload', 'sourcesUploadedTitle')}
+                contentClassName={styles.files}
+                sub
+            >
+                <FilesSearchInput
+                    className={styles.search}
+                    options={files}
+                    onSearchFiles={handleSearchFiles}
+                />
```

Comment:
The search state should be moved to this parent component and parent should implement the filtering logic